drivers/lora/sx12xx_common.c

```diff
@@ -331,6 +376,98 @@ int sx12xx_lora_recv_async(const struct device *dev, lora_recv_cb cb)
 	return 0;
 }
 
+int sx12xx_lora_cad(const struct device *dev, k_timeout_t timeout)
+{
+	/* Validate that we have a CAD configuration */
+	if (!dev_data.tx_cfg.datarate) {
+		return -EINVAL;
+	}
+
+	struct k_poll_signal done = K_POLL_SIGNAL_INITIALIZER(done);
+	struct k_poll_event evt =
+		K_POLL_EVENT_INITIALIZER(K_POLL_TYPE_SIGNAL, K_POLL_MODE_NOTIFY_ONLY, &done);
+	int ret;
+	bool detected = false;
+
+	/* Ensure available, decremented by sx12xx_ev_cad_done or on timeout */
+	if (dev_data.async_rx_cb) {
+		/* Increment atomic so both acquire and release will fail */
+		if (!atomic_cas(&dev_data.modem_usage, STATE_BUSY, STATE_CLEANUP)) {
+			return -EBUSY;
+		}
+		/* Put radio back into sleep mode */
+		Radio.Sleep();
+		/* Completely release modem */
+		dev_data.operation_done = NULL;
+		atomic_clear(&dev_data.modem_usage);
```

Comment:
This looks exactly the same as `modem_release()`. Why was it copied here? Can't the function be called?